src/App.jsx

```diff
@@ -14,33 +15,105 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+export const App = () => {
+  const [sort, setSort] = useState([...goodsFromServer]);
+  const [reverse, setReverse] = useState(false);
+  const [active, setActive] = useState('');
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  const handle = good => {
+    let newSort;
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+    switch (good) {
+      case 'abc':
+        if (reverse) {
+          newSort = [...sort].sort((a, b) => b.localeCompare(a));
+          setActive(good);
+        } else {
+          newSort = [...sort].sort((a, b) => a.localeCompare(b));
+          setActive(good);
+        }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+        break;
+      case 'length':
+        if (reverse) {
+          newSort = [...sort].sort((a, b) => b.length - a.length);
+          setActive(good);
+        } else {
+          newSort = [...sort].sort((a, b) => a.length - b.length);
+          setActive(good);
+        }
+
+        break;
+      case 'reverse':
+        if (active === 'abc') {
+          setReverse(!reverse);
+        } else if (active === 'reverse') {
+          setReverse(!reverse);
+          setActive('');
+        } else if (active === 'length') {
+          setReverse(!reverse);
+        } else {
+          setReverse(!reverse);
+          setActive('reverse');
+        }
+
+        newSort = [...sort].reverse();
+        break;
+      default:
+        setActive('');
+        setReverse(false);
+        newSort = [...goodsFromServer];
+    }
+
+    setSort(newSort);
+  };
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={`button is-info ${(active !== 'abc' && 'is-light') || (active !== 'abc' && reverse ? 'is-light' : '')}`}
+          onClick={() => handle('abc')}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={`button is-success ${(active !== 'length' && 'is-light') || (active !== 'length' && reverse) ? 'is-light' : ''}`}
```

Comment:
Similar to the previous comment, the condition `(active !== 'length' && 'is-light') || (active !== 'length' && reverse) ? 'is-light' : ''` is flawed. It will always apply 'is-light' when `active` is not 'length'. Simplify this logic to ensure it accurately represents the button's active state.